libp2p/protocols/kademlia/dhttypes.nim

```diff
@@ -0,0 +1,46 @@
+import chronicles
+import std/[tables, hashes]
+import results
+
+type EntryKey* = object
+  data*: seq[byte]
+
+type EntryVal* = object
+  data*: seq[byte]
+
+type TimeStamp* = object
+  # Currently a string, because for some reason, that's what is chosen at the protobuf level
+  # TODO: convert between RFC3339 strings and use of integers (i.e. the _correct_ way)
+  ts*: string
+
+type EntryCandidate* = object
+  key*: EntryKey
+  val*: EntryVal
+  time*: TimeStamp
+
+type ValidatedEntry* = object
+  key*: EntryKey
+  val*: EntryVal
+  time*: TimeStamp
+
+type EntryValidator* = ref object of RootObj
+method validate*(
+    self: EntryValidator, key: EntryKey, val: EntryVal
+): bool {.base, raises: [], gcsafe.} =
+  discard
```

Comment:
Consider adding an assertion here, so it forces the dev to implement this method

Reply:
resolved: [36533d3](https://github.com/vacp2p/nim-libp2p/pull/1582/commits/36533d3eb1409da7566c40aa1c90f96ae78ddde5)

(let me know if there's a standard format to fail with)